python/lsst/obs/base/ingest_tests.py

```diff
@@ -255,9 +274,11 @@ def testDirect(self):
         # Check that it really did have a URI outside of datastore
         srcUri = ButlerURI(self.file)
         butler = Butler(self.root, run=self.outputRun)
-        datasets = list(butler.registry.queryDatasets("raw", collections=self.outputRun))
+        datasets = list(butler.registry.queryDatasets(self.ingestDatasetTypeName, collections=self.outputRun))
         datastoreUri = butler.getURI(datasets[0])
-        self.assertEqual(datastoreUri, srcUri)
+        # These are known to be file URIs but sometimes we can end up with
+        # scheme-less and file and they won't compare
```

Comment:
"scheme-less and file"?

I assume this is attached to the `assertEqual` below: is it implying that the test will fail sometimes under particular conditions? Could those conditions be trapped for?

Reply:
I think that I fixed the issue in lsst/daf_base#485 so this can be a straight equality now.

Reply:
Actually, it's triggered by running the test from pytest vs python since I was not forcing an absolute path in the test directory calculation. I will fix that.